### PR TITLE
refactor: async/await

### LINKED ISSUE #31: Native async/await
- [x] Native async/await
  * [x] Remove callback
  * [x] Utilize [request-promise-native](https://www.npmjs.com/package/request-promise-native)
  * [x] Replace stream with string
    * [x] Replace fs.createReadStream() with fs.readFile() of hexo-fs
    * [x] Replace feedparser with [feed-furious](https://github.com/tuananh/feed-furious) to support string input, instead of stream.
      * [x] Port feed-furious to support async-ed camaro v4.
- [x] IDN support https://github.com/hexojs/hexo-migrator-rss/pull/32

### PR DESCRIPTION
**Pull request by curbengh**
Closes #31 

BREAKING CHANGE: Drops Node 8
[got v10](https://github.com/sindresorhus/got/releases/tag/v10.0.0) requires Node 10

### CODE REVIEW

**Comment by curbengh** [comment]
Unit test for `alias` option and atom feed to be done in another PR.

File: lib/migrator.js [hunk 1]
```diff
@@ -1,90 +1,88 @@
...
+        log.i('Post found: %s', item.title);
+      }
 
-        posts.push(newPost);
+      const newPost = {
+        title: item.title,
+        date: item.date,
+        tags: item.categories,
+        excerpt: tomd.turndown(item.description || ''),
+        content: tomd.turndown(item.content || item.description || '')
```
Comment (curbengh) [inline lib/migrator.js]: plan to implement a feature to request from the post's url if item.content is absent, in a separate PR.

File: lib/migrator.js [hunk 2]
```diff
@@ -1,90 +1,88 @@
...
-        if (args.alias) {
-          newPost.alias = url.parse(item.link).pathname;
-        }
+    feed = await parseFeed(input);
+  } catch (err) {
+    throw new Error(err);
+  }
+
+  if (feed) {
+    for (const item of feed.items) {
```
Comment (curbengh) [inline lib/migrator.js]: the loop here can be utilized to limit new posts (#9), to do in future PR.

**Comment by curbengh** [comment]
Tested with a dummy rss2.xml.

File: lib/migrator.js [hunk 3]
```diff
@@ -1,90 +1,88 @@
...
-const request = require('request');
-const url = require('url');
-const fs = require('fs');
-
-exports.registerMigrator = function(hexo) {
-  hexo.extend.migrator.register('rss', (args, callback) => {
-    const source = args._.shift();
-    const tomd = new TurndownService();
-
+const request = require('request-promise-native');
```
Comment (SukkaW) [inline lib/migrator.js]: Maybe `got` or `node-fetch` could be considered?

Comment (curbengh) [reply lib/migrator.js]: I didn't realize request is under [maintenance mode](https://github.com/request/request/issues/3142) (only knew it through [`got`](https://www.npmjs.com/package/got)). `got` looks promising, I'll have a look.

File: lib/migrator.js [hunk 4]
```diff
@@ -1,90 +1,88 @@
...
     }
 
-    const log = hexo.log;
-    const post = hexo.post;
-    let untitledPostCounter = 0;
-    let stream;
-
     if (/^http(s)?:\/\//i.test(source)) {
-      stream = request(source);
+      input = await got(source, { resolveBodyOnly: true, retry: 0 });
```
Comment (curbengh) [inline lib/migrator.js]: `retry` is disabled to pass the unit test (in local machine with slow connection).
The `got` options can be exposed to config in a separate PR.

File: lib/migrator.js (hunk 3 above)
Comment (curbengh) [reply lib/migrator.js]: Refactored to `got`.
